src/System.Slices/System/SpanExtensions_binary.cs

```diff
@@ -58,7 +58,37 @@ public static partial class SpanExtensionsLabs
             where T : struct
         {
             Contract.RequiresInInclusiveRange(Unsafe.SizeOf<T>(), (uint)slice.Length);
-            return slice.Cast<byte, T>()[0];
+            return Unsafe.As<byte, T>(ref slice.DangerousGetPinnableReference());
+        }
+
+        /// <summary>
+        /// Reads a structure of type T out of a slice of bytes.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool TryRead<[Primitive]T>(this ReadOnlySpan<byte> slice, out T value)
```

Comment:
It does not do anything yet. We have plans to make it into a constraint that would match value types that have no references in them.